## Answers


1. Cross and Birley (1986) studied restriction site vvariation in the region of alccohol dehydrogenase (*Adh*)gene of *D. melanogaster* ina  population descended from flies trapped at a Dutch fruit market in Groningen.  The following data was collected:

| *Adh* Allele | *Eco*RI Allele| Individuals|
|-------------|----------------|------------|
*Adh<sup>F</sup>*| *Eco*RI<sup>+</sup> | 22|
*Adh<sup>S</sup>*| *Eco*RI<sup>+</sup> | 4|
*Adh<sup>F</sup>*| *Eco*RI<sup>-</sup> | 3|
*Adh<sup>S</sup>*| *Eco*RI<sup>-</sup> | 5|

Where *Adh<sup>F</sup>* and *Adh<sup>S</sup>* are the allozyme alleles of *Adh* and *Eco*RI<sup>+</sup> and *Eco*RI<sup>-</sup> indicate the presence or absence of an *Eco*RI restriction site 3.5 kb downstream

   * Test for the presence of linkage disequilibrium
   
AB = G1 = (p1) (p2)

Ab = G2 = (p1) (q2)

aB = G3 = (q1) (p2) 

ab = G4 = (q1) (q2)

p1 and q1 are the frequency of the alleles

AB= F+

Ab= F-

aB= s+

ab= s-


p1= F frequency = 25/34 = 0.735

q1= s frequency = 9/34 = 0.265

P2= + frequency = 26/34 = 0.765

q2= - frequency = 8/34 = 0.235



G1= 0.562275

G2= 0.172725

G3= 0.202725

G4= 0.062275


D = (G1 G4) − (G2 G3)

D = (0.035) - (0.035)

D = 0

   * If significant, express *D* relative to its theoretical maximum or minimum.
  
Since D = 0 the population is at equilibrium and is not significant.


2. Three linked loci in a random mating population have the gametic frequencies shown in the table below. Calculate the linkage disequilibrium D/Dmax between each pair of genes. Explain why the results seem paradoxical.

|Genotype | Frequency|
|---------|------|
| *ABC*| 0.25|
| *ABc*| 0|
| *AbC*| 0.25|
|*Abc*| 0|
|*aBC* | 0|
|*aBc* | 0.25|
|*abC* | 0|
|*abc*|0.25|


D'=D/Dmax


AB Calculations

D=(G1G4)-(G2G3)

D=(0.25* 0.25)-(0.25 *0.25)

D/Dmax=0/0.25=0


BC calculations

D=(G1G4)-(G2G3)

D=(0.25* 0.25)-(0.25 *0.25)

D/Dmax=0/0.25=0



AC calculations

D=(G1G4)-(G2G3)

D=(0.5* 0.5)-(0 *0)

D=0.25

D/Dmax=0.25/0.25

D' = 1

This seems paradoxical because two of the calculations of alleles at this loci are saying that this is in equilibrium, but the third is very far from equilibrium. 

3. Two populations are examined for the same pair of loci with the following results in terms of two-locus gamete numbers:

|Population|Gamete| Count|
|----------|------|-------|
| Pop 1| AB | 720|
| Pop 1| Ab | 180|
| Pop 1| aB | 80|
| Pop 1| ab | 20|
| Pop 2| AB | 300|
| Pop 2| Ab | 2700|
| Pop 2| aB | 700|
| Pop 2| ab | 6300|

These two populations are now combined to form a third:

|Population|Gamete| Count|
|----------|------|-------|
| Pop 3| AB | 1020|
| Pop 3| Ab | 2880|
| Pop 3| aB | 780|
| Pop 3| ab | 6320|

   * Calculate the two-locus gamete frequencies for each population and the gameticphase imbalance (linkage disequilibrium). Is any population out of linkage equilibrium?
 
|Population|Gamete| Count| Gamete Frequency | 
|----------|------|-------|-----------------|
| Pop 1| AB | 720| 0.72|
| Pop 1| Ab | 180| 0.18|
| Pop 1| aB | 80| 0.08|
| Pop 1| ab | 20| 0.02|
| Pop 2| AB | 300| 0.03|
| Pop 2| Ab | 2700| 0.27|
| Pop 2| aB | 700| 0.07|
| Pop 2| ab | 6300| 0.63|
| Pop 3| AB | 1020| 0.0927|
| Pop 3| Ab | 2880| 0.262|
| Pop 3| aB | 780| 0.071|
| Pop 3| ab | 6320| 0.575|

Population 1 Linkage Disequilibrium

D=(G1G4)-(G2G3)

D=(0.72 * 0.02) - (0.18 * 0.08)

D= (0.0144) - (0.0144)

D=0

The population is in equilibrium.


Population 2 Linkage Disequilibrium

D=(G1G4)-(G2G3)

D=(0.03 * 0.63) - (0.27 * 0.07)

D= (0.0189) - (0.0189)

D=0

The population is in equilibrium.


Population 3 Linkage Disequilibrium

D=(G1G4)-(G2G3)

D= (0.0927 * 0.575) - (0.262 * 0.071)

D= (0.0533) - (0.0186)

D=0.0347

This population is in disequilibrium.

   * What is the effect of pooling of the two populations on linkage equilibrium?

By pooling the two populations, which were in linkage equilibrium, it makes the new combined population in linkage disequilibrium.

   * Suppose that the recombination rate between loci A/a and B/b is 0.1. What is the expected disequibrium in the offspring of population 3 (the parental gene pool), assuming random mating? In the second generation?
 

D' = D(1−r) where r is the recombination rate

r=0.1

D=0.0347

D' = 0.0347 (1-0.1)

D' = 0.03123


Second generation:

D' = D(1−r) where r is the recombination rate

r=0.1

D=0.03123

D' = 0.03123 (1-0.1)

D'= 0.028107

   * Can this single episode of admixture be detected in the population established from population 3 after two generations of random mating? Can it be detected in the genotype frequencies at the A/a locus after two generations of random mating? Can it be detected in the genotype frequencies at the B/b locus after two generations of random mating?

The admixture can still be detected in population 3 after two generations of random mating because the population is still in linkage disequilibrium. This could still be detected at the A/a and B/b locus.

4. A continuous trait is distributed approximately according to a normal distribution with mean 100 and standard deviation of 15.  What proportion of the population is expected to have a phenotypic value above 130?  Below 85?  Above 85?

**Not sure how to do this**

5. A genetically heterogeneous population of wheat has a variance in the number A days to maturation of 40, whereas two inbred populations derived from it have a variance in the number of days to maturation of 10.  
   * What is the genotypic variance (V<sub>G</sub> or $\sigma_{g}^2$), the environmental variance (V<sub>E</sub> or $\sigma_{e}^2$), and the broad sense heritablilty *H*<sup>2</sup>, of days to maturation in this population?

Vp=V<sub>E</sub> + V<sub>G</sub>


So V<sub>E</sub> would be the value of 10 because those are the inbred lines. 
V<sub>P</sub> would be 40 because that is the population value. 


40=10 + V<sub>G</sub>

30 = V<sub>G</sub>


*H*<sup>2</sup> = V<sub>G</sub> / V<sub>P</sub>

*H*<sup>2</sup> = 30/40

*H*<sup>2</sup> = 0.75

   * If the inbred lines were crossed, what would be the predicted variance in days to maturation of the F<sub>1</sub> generation?

If the inbred lines were crossed, the generation would still be the same because they are inbred, so they would be genetically the same. So it would still be 10 because that is the value of the parents.

6. In comparing a quantitative trait in the F<sub>1</sub> and F<sub>2</sub> generations obtained by crossing two highly inbred strains: 

   * Which set of progeny provides an estimate of the environmental variance?
 

The environmental variance in inbred generations would be the same in the F<sub>1</sub> generation, so that would give the best estimate.

   * What determines the variance of the other set of progeny?

Any other progeny (ex. the F<sub>2</sub> generation would return to the Vp=V<sub>E</sub> + V<sub>G</sub> formula, so both the environmental and genetic variance would be used to determine population variance)

7. In a cross between two cultivated inbred varieties of tobacco, the variance in leaf number per plant in the F<sub>1</sub> generation is 1.46 and in the F<sub>2</sub> generation it is 5.97. What are the genotypic and environmental variances? What is the broad-sense heritability in leaf number?


Since the populations are inbred, F<sub>1</sub> is the V<sub>E</sub>, so V<sub>E</sub> = 1.46.

Vp=V<sub>E</sub> + V<sub>G</sub>

5.97 = 1.46 + V<sub>G</sub>

V<sub>G</sub> = 4.51



H<sub>B</sub> = V<sub>G</sub> / V <sub>P</sub>

H<sub>B</sub> = 4.51 / 5.97

H<sub>B</sub> = 0.7554



