# Chapter 9: Population Subdivision
#### Notes taken by Keegan Hart

## Variation within species exsits on two primary levels
* Fis = genetic variation within local pops
* Fst = genetic varation among local 

## Fis
* measure of departure from HW proportions within a population

## Shrub case study
* 6% offspring selfing and 40% random mating doesnt = 100 why?
** probably a typo and should be 60

## Wahlund Effect
* fewer heterozygotes compared to HWE due to subdivision into demes
* if you broke it up and looied at each deme individually they would all be within HWE but as a whole they do not seem to be in HWE
* Trout example:
* both groups were nearly fixed for two alleles
heterozygosity was 73% lower than if there was random mating among all members of lake 
* emphasizes you need to be careful about how you sample to insure you account for subpopulations

## Excess of Homozygotes
* Fis > 0
* caused bu non-random matings and poulation dubdivision - inbreeding

## Excess of Heterozygotes
* fis < 0 
* can be caused by 
small effective pop size
etc...

## SGS = Spatial Genetic Structure
* how genetics are structured spacially 
* spacial genetic structure
* Fij = coancestry coefficient = probability that a random allele from i is identical to random allele from j
* Q= prob that two alleles are identical
* r= spatial distance between them 
* some examples of SGS with benthic invertebrates with short-lived pelagic larvae 

## complete isolation vs gene flow
* 9.4 illustrates important principle of how we understand fst
* what is takeaway?
* small effective population size -> fst increases rapidly over time because of drift 
* small populations you esssentially have more of an impact on stochasitc variation so they tend to drift apart much more frequently 

## Island Model
* never makes sense!!!
* difference between island and stepping stone?
* island is one of the situations where it is theoretical you only way you get a neat relationship is when every subpop is equal in size and equally likely to exchange genetic informaiton which any subpop
* fst is built off of this situation^^^
* any time you are looking for effective migrants it is based on the island model 

## Stepping Stone
* still pretty unrealistic 
* individual subpops with same # indivs and still equal migration rate but difference is certain subpops can only migrate to other specific subpop (can only get to nearest neighbor)
* still not that realistic 
* attempt to model isolation by distance 

## continuously distributed pops
* neighborhood = area from which indivs can be considered to be drawn at random from random mating pop
* no distinct bounderies betweeen groups 
* nearby indivs are more likely to mate with those that are closer than those that are far away 
* no mate selection assumption

## cytoplasmic genes and sex-linked markers
* more differentiation in mitochondrial and chloroplast DNA due ot them being passed down from one parent 
* effective population size of nuclear dna is greater than effective population size of mitochondrial and chloroplast 
* figure 9.11 is one of few ways we have of relating fst and gettign a meaningful value of migration rate * population size (mN) 
* if we are looking at one locus recombination doesnt matter it is based solely on effective population size
* if we are compating one mutation in mito vs one in nuclear we have to start factoring in recombination 

## Limitations of Fst
* this section is very important
* figures 9.14 and 9.15 
* any pop genetics paper they are still talking about fst but theres lots of limitations 
* high levels of heterozygosity and why that is problematic with fst 
* if you have really high het you wont be able to detect distinct populations 
* depending on het that is present at locus the higher the het the lower your max fst was you cant get to 1 anymore 
* gst = multilocus analog fst
* gst' = 
* josts is not true analog of fst it is more a measure of genetic differentiation -> genetic distance 
* fst is not genetic distance* 
* migration drift equilibrium = theoretically theres an assumption that migration rate isnt changing much through time and that there is a balance between drift and migration because of that, however after last big ice age nearly every taxa has been expanding 
* is an expanding population in migration drift equilibrium? no so basically every population has likely violated the migration drift equilibrium 
* fst is horribel way at getting at gene flow so its used because its somehwat standardized and easy to calculate
* MLE have become more in favor as far as looking at levels of migration through coalescence based analysis 
* start to look at geneological relationship between different alleles and form a phylogenetic tree back thorugh time and you are abel to use diferent statistics to determine igratio rate and effective population size - takes really long time 
* phylogenetic methods dont rely on migration drift equilibrium 
* mN = avg rate over huge swaths of evolutionary time 
* marking and tagging gives you a better sense of what is happening right now 

## estimation of gene flow 
* private allele = allele only foudn in one population
* how is that informative for migration?
* you can see how that allele shows up in other populations to determine migration rate 

# 