#Week 12
##Part 1 Simulated Data
For this part of the assignment I used the filtered vcf file generated in the Week 11 assignment, all steps I took to make the filtered file can be found there. The file name of the vcf I will be using is "FINAL.vcf".
###Converting vcf using PGDSpider
```{bash}
cp /home/BIO594/DATA/Week7/example/BSsnp.spid .
ln -s ../popmap .
mamba install pgdspider
PGDSpider2-cli -inputfile FINAL.vcf -outputfile SNP.TRSdp5p05FHWEBS -spid BSsnp.spid
```

###Outlier Detection using Bayescan
```{bash}
BayeScan2.1_linux64bits SNP.TRSdp5p05FHWEBS -nbp 30 -thin 20
```
In Rstudio: 
```{r}
cat(plot_bayescan("SNP.TRSdp5p05FH_fst.txt")[[1]][1], file = "Baye_outliers.txt")
```
Bayescan found zero outliers.

###Outlier Detection using PCA
Removing SNPs with more than 2 alleles
```{bash}
vcftools --vcf FINAL.vcf --max-alleles 2 --recode --recode-INFO-all --out SNP.TRSdp5p05FHWE2A
```

The following code was placed into an R file before running
```{r}
library(pcadapt)

#load our VCF file into R
filename <- read.pcadapt("SNP.TRSdp5p05FHWE2A.recode.vcf", type = "vcf" )

#Create first PCA
x <- pcadapt(input = filename, K = 20)

#Plot the likelihoods
plot(x, option = "screeplot")
#Plot Plot the likelihoods for only first 10 K
plot(x, option = "screeplot", K = 10)

#Create population designations
poplist.names <- c(rep("POPA", 20),rep("POPB", 20),rep("POPC", 20), rep("POPD",20))

#Plot the actual PCA (first two PCAs)
plot(x, option = "scores", pop = poplist.names)
#Plot PCA with PCA 2 and PCA 3
plot(x, option = "scores", i = 2, j = 3, pop = poplist.names)
#Plot PCA with PCA 3 and PCA 4
plot(x, option = "scores", i = 3, j = 4, pop = poplist.names)

#Redo PCA with only 3 K
x <- pcadapt(filename, K = 3)

summary(x)

#Start looking for outliers
#Make Manhattan Plot
plot(x , option = "manhattan")
#Make qqplot
plot(x, option = "qqplot", threshold = 0.1)
# Look at P-value distribution
plot(x, option = "stat.distribution")

# Set FDR
library(qvalue)
qval <- qvalue(x$pvalues)$qvalues
alpha <- 0.1

# Save outliers
outliers <- which(qval < alpha)


# Testing for library effects

poplist.names <- c(rep("LIB1", 40),rep("LIB2", 40))
x <- pcadapt(input = filename, K = 20)

plot(x, option = "scores", pop = poplist.names)
plot(x, option = "scores", i = 2, j = 3, pop = poplist.names)



x <- pcadapt(filename, K = 2)

summary(x)

plot(x , option = "manhattan")
plot(x, option = "qqplot", threshold = 0.1)

plot(x, option = "stat.distribution")

library(qvalue)
qval <- qvalue(x$pvalues)$qvalues
alpha <- 0.1
outliers <- which(qval < alpha)
write.table(outliers, "PCA_outliers.txt", row.names=FALSE)
```
PCA Adapt found outliers at 149 and 150

###Combining outliers from both analyses
```{bash}
mawk '!/#/' FINAL.vcf | cut -f1,2 > totalloci
NUM=(`cat totalloci | wc -l`)
paste <(seq 1 $NUM) totalloci > loci.plus.index
cat PCA_outliers.txt | parallel "grep -w ^{} loci.plus.index" | cut -f2,3> outlier.loci.txt

```

###Creating a vcf file with neutral loci
```{bash}
vcftools --vcf FINAL.vcf --exclude-positions outlier.loci.txt --recode-INFO-all --out neutralloci --recode
```

###Running PCA Analysis, DAPC, Isolation by distance and pairwise Fst with neutral loci
```{r}
library(adegenet)
library(vcfR)


my_vcf <- read.vcfR("neutralloci.recode.vcf")


my_genind <- vcfR2genind(my_vcf)


strata<- read.table("strata", header=TRUE)
strata_df <- data.frame(strata)
strata(my_genind) <- strata_df

setPop(my_genind) <- ~Population

#Test Population Structure
library(hierfstat)
wc(my_genind)
matFst <- pairwise.WCfst(my_genind)



#PCA

X <- tab(my_genind, freq = TRUE, NA.method = "mean")
pca1 <- dudi.pca(X, scale = FALSE, scannf = FALSE, nf = 3)
barplot(pca1$eig[1:50], main = "PCA eigenvalues", col = heat.colors(50))
s.class(pca1$li, pop(my_genind))
title("PCA of simulated dataset\naxes 1-2")
add.scatter.eig(pca1$eig[1:20], 3,1,2)

col <- funky(15)
s.class(pca1$li, pop(my_genind),xax=1,yax=2, col=col, axesell=FALSE, cstar=0, cpoint=3, grid=FALSE)

#DAPC

grp <- find.clusters(my_genind, max.n.clust=40)
table(pop(my_genind), grp$grp)

table.value(table(pop(my_genind), grp$grp), col.lab=paste("inf", 1:2), row.lab=paste("ori", 1:4))


dapc1 <- dapc(my_genind, grp$grp)
scatter(dapc1,col=col,bg="white", solid=1)



contrib <- loadingplot(dapc1$var.contr, axis=1, thres=.01, lab.jitter=1)
contrib


setPop(my_genind) <- ~Library

dapc1 <- dapc(my_genind, pop(my_genind))
contrib <- loadingplot(dapc1$var.contr, axis=1, thres=.05, lab.jitter=1)

# Isolation by distance
my_vcf <- read.vcfR("neutralloci.recode.vcf")


my_genind <- vcfR2genind(my_vcf)


strata<- read.table("strata", header=TRUE)
strata_df <- data.frame(strata)
strata(my_genind) <- strata_df

setPop(my_genind) <- ~Population

xy <-read.table("dist.mat")
xy
my_genind@other$xy <- xy


toto <- genind2genpop(my_genind)
Dgeo <- dist(my_genind$other$xy)
Dgen <- dist.genpop(toto,method=2)
ibd <- mantel.randtest(Dgen,Dgeo)

plot(ibd)
plot(Dgeo, Dgen)
abline(lm(Dgen~Dgeo), col="red",lty=2)


```
